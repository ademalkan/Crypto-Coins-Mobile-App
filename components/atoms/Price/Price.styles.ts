import { StyleSheet } from "react-native";

export default StyleSheet.create({
  price: {
    fontSize: 16,
    marginLeft: 'auto',
    fontWeight: 'bold',
  },
})