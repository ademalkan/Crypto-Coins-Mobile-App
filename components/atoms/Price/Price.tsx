import React from 'react';
import { View, Text, Image } from 'react-native';
import styles from "./Price.styles";

type PriceProps = {
  price: number;
};

const Price: React.FC<PriceProps> = ({ price }) => {
  return (
    <Text style={styles.price}>${price}</Text>
  );
};

export default Price;
