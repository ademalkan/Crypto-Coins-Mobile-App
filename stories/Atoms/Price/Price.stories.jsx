import React from "react";
import { View } from "react-native";
import Price from "../../../components/atoms/Price/Price";

export default {
  title: "Atoms/Price",
  component: Price,
  args: {
    price: 10.99,
  },
  decorators: [
    (Story) => (
      <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
        <Story />
      </View>
    ),
  ],
};

export const Default = ({ ...args }) => <Price {...args} />;
