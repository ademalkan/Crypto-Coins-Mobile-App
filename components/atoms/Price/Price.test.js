import React from "react";
import renderer from "react-test-renderer";
import Price from "./Price";

describe("Price", () => {
  it("should render the price prop correctly", () => {
    const price = 9.99;

    const component = renderer.create(<Price price={price} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
