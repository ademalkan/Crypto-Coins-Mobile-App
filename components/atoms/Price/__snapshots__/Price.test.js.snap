// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should render the price prop correctly 1`] = `
<Text
  style={
    {
      "fontSize": 16,
      "fontWeight": "bold",
      "marginLeft": "auto",
    }
  }
>
  $
  9.99
</Text>
`;
